### PR TITLE
Make the tool gpg-like with its cli flags

### PR DESCRIPTION
**Pull request by ncdc**
Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

### CODE REVIEW

**Comment by ncdc** [comment]
Looks like I need to add `--clearsign` support for deb-s3. Hang tight 😄

File: main.go [hunk 1]
```diff
@@ -33,60 +39,263 @@ var (
...
-	GOOGLE_APPLICATION_CREDENTIALS (optional):
-		Path to Google Cloud credentials file.
-	PGP_UID_NAME (generate only):
-		Name for generated PGP key (e.g., "Phil Zimmermann").
-	PGP_UID_COMMENT (generate only, optional):
-		Comment for generated PGP key.
-	PGP_UID_EMAIL (generate only):
-		Email for generated PGP key (e.g., "phil@example.com")
-`, os.Args[0])
+	pflag.Usage()
```
Comment (mattmoyer) [inline main.go]: Maybe we could add a little extra usage blurb beyond what pflag returns here? I couldn't figure how to use `--export` correctly. Maybe add a message like `func usage(msg string) {` for clarity?

**Review by mattmoyer — COMMENTED** [review]
Overall +1, will review again with the extra changes in just a sec.

File: main.go (hunk 1 above)
Comment (ncdc) [reply main.go]: Sure

**Comment by ncdc** [comment]
@mattmoyer updated with clearsign support and slightly better usage output. One behavior that doesn't match `gpg` 100% - if you do `--clearsign --detach-sign`, because of the way my switch statement is coded, it will do a detached signature, but it won't clearsign it. The only way to clearsign is with non-detached signatures currently.

**Review by mattmoyer — APPROVED** [review]

**Review by mattmoyer — APPROVED** [review]

**Review by mattmoyer — APPROVED** [review]
Thanks @ncdc!

**Pull request merged**